intercept/src/intercept.cpp

```diff
@@ -7244,13 +7304,212 @@ void CLIntercept::setKernelArgUSMPointer(
     }
 }
 
+void CLIntercept::dumpKernelSource(cl_kernel kernel, 
+                                   uint64_t enqueueCounter,
+                                   bool byKernelName )
+{
+    std::lock_guard<std::mutex> lock(m_Mutex);
+    std::string fileNamePrefix = "";
+    OS().GetDumpDirectoryName( sc_DumpDirectoryName, fileNamePrefix );
+    fileNamePrefix += "/Replay/Enqueue_";
+    if (byKernelName)
+        fileNamePrefix += getShortKernelName(kernel);
+    else
+        fileNamePrefix += std::to_string(enqueueCounter);
+    fileNamePrefix += "/";
+    OS().MakeDumpDirectories( fileNamePrefix );
+
+    // Get the cl_program from the cl_kernel, then extract kernel source code from the cl_program
+    CLIntercept* pIntercept = GetIntercept();
+
+    cl_program tmp_program;
+    pIntercept->dispatch().clGetKernelInfo(kernel, CL_KERNEL_PROGRAM, sizeof(cl_program), &tmp_program, nullptr);
+
+    try
+    {
+        std::string source = m_SourceStringMap.at(tmp_program);
```

Comment:
I'm guessing the exception handler is intended to handle the case where the program is not in the source string map?

It might be clearer to just check for this instead.

Reply:
(Can we pretend that I never wrote this piece of code?)

(fixed)